flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -538,16 +538,16 @@ public CatalogBaseTable getTable() {
 
     /**
      * Returns an array of names of all views(both temporary and permanent) registered in the
-     * namespace of the current catalog and database.
+     * namespace of the given catalog and database.
      *
      * @return names of registered views
      */
     public Set<String> listViews(String catalogName, String databaseName) {
-        Catalog currentCatalog = catalogs.get(getCurrentCatalog());
+        Catalog catalog = catalogs.get(catalogName);
 
         try {
             return Stream.concat(
-                            currentCatalog.listViews(getCurrentDatabase()).stream(),
+                            catalog.listViews(databaseName).stream(),
```

Comment:
Can we also fix the exception message a few lines below in case the database doesn't exist?

Reply:
Done, Thanks for your comment 👍